ui/server/src/test/scala/pl/touk/nussknacker/ui/api/ProcessesResourcesSpec.scala

```diff
@@ -233,27 +237,36 @@ class ProcessesResourcesSpec extends FunSuite with ScalatestRouteTest with Match
     val thirdProcessor = ProcessName("Processor3")
 
     createProcess(firstProcessor, testCategoryName, false)
-    createProcess(secondProcessor, testCategoryName, false)
+    createDeployedCanceledProcess(secondProcessor, testCategoryName, false)
     createDeployedProcess(thirdProcessor, testCategoryName, false)
 
     Get(s"/processes") ~> routeWithAllPermissions ~> check {
       status shouldEqual StatusCodes.OK
-      val data =responseAs[List[Json]]
+      val data = responseAs[List[Json]]
       data.size shouldBe 3
+
+      val process = findJsonProcess(responseAs[String], firstProcessor.value)
+      process.flatMap(_.stateStatus) shouldBe Some(SimpleStateStatus.NotDeployed.name)
     }
 
     Get(s"/processes?isDeployed=true") ~> routeWithAllPermissions ~> check {
       status shouldEqual StatusCodes.OK
       responseAs[List[Json]].size shouldBe 1
+
       val process = findJsonProcess(responseAs[String], thirdProcessor.value)
-      process.flatMap(_.name) shouldBe Some(thirdProcessor.value)
+      process.map(_.name) shouldBe Some(thirdProcessor.value)
+      process.flatMap(_.stateStatus) shouldBe Some(SimpleStateStatus.Running.name)
     }
 
     Get(s"/processes?isDeployed=false") ~> routeWithAllPermissions ~> check {
       status shouldEqual StatusCodes.OK
       responseAs[List[Json]].size shouldBe 2
+
       val process = findJsonProcess(responseAs[String], thirdProcessor.value)
-      process.flatMap(_.name) shouldBe Option.empty
+      process.map(_.name) shouldBe Option.empty
+
+      val canceledProcess = findJsonProcess(responseAs[String], secondProcessor.value)
+      canceledProcess.flatMap(_.stateStatus) shouldBe Some(SimpleStateStatus.Canceled.name)
```

Comment:
can you use: `canceledProcess.value.stateStatus.value...`? or even split it to two lines:
```
val canceledProcess = findJsonProcess(responseAs[String], secondProcessor.value).value
canceledProcess.stateStatus.value...
```
(In all places where you used flatMap in checks)
Using current syntax after fail you can't recognize what was wrong (two checks are flatten to one)

Reply:
Great thinks - done.